tests/core/p2p-proto/test_les_api.py

```diff
@@ -0,0 +1,129 @@
+import asyncio
+
+import pytest
+
+from eth.chains.base import MiningChain
+from eth.tools.builder.chain import (
+    build,
+    disable_pow_check,
+    genesis,
+    latest_mainnet_at,
+    mine_block,
+)
+
+from trinity.db.eth1.header import AsyncHeaderDB
+from trinity.protocol.les.api import LESAPI
+from trinity.protocol.les.commands import Announce
+from trinity.protocol.les.handshaker import LESHandshakeReceipt
+from trinity.protocol.les.proto import LESProtocolV2
+from trinity.tools.factories import (
+    ChainContextFactory,
+    LESV2PeerPairFactory,
+)
+
+
+@pytest.fixture
+def bob_chain():
+    chain = build(
+        MiningChain,
+        latest_mainnet_at(0),
+        disable_pow_check(),
+        genesis(),
+    )
+    return chain
+
+
+@pytest.fixture
+def alice_headerdb(bob_chain):
+    bob_genesis = bob_chain.headerdb.get_canonical_block_header_by_number(0)
+
+    chain = build(
+        MiningChain,
+        latest_mainnet_at(0),
+        disable_pow_check(),
+        genesis(params={"timestamp": bob_genesis.timestamp}),
+    )
+    return AsyncHeaderDB(chain.headerdb.db)
+
+
+@pytest.fixture
+async def alice_and_bob(alice_headerdb, bob_chain):
+    pair_factory = LESV2PeerPairFactory(
+        alice_client_version='alice',
+        alice_peer_context=ChainContextFactory(headerdb=alice_headerdb),
+        bob_client_version='bob',
+        bob_peer_context=ChainContextFactory(headerdb=AsyncHeaderDB(bob_chain.headerdb.db)),
+    )
+    async with pair_factory as (alice, bob):
+        yield alice, bob
```

Comment:
It took me a little while to get bob and alice straight in my head, then I *think* I realized what it is. `alice` here, is the `peer` object that would be on "Alice's node". From Alice's perspective, the peer object represents Bob.

Does that sound right? If so, I'm not really sure what to do about it to make that clearer. This might have helped:
```suggestion
        yield alices_bob, bobs_alice
```

Or maybe `to_bob, to_alice`, or even just reversing them. :man_shrugging:

Reply:
Probably not the answer you're looking for but it kind of doesn't matter who's who.  Just that one is one, and the other is the other....  `alice/bob` are easier to keep mentally separated than `peer_a/peer_b`, etc.

Reply:
Let me see if I can point to a specific place it tripped me up. (and why I think it matters)